pkg/bindings/stan/message.go

```diff
@@ -0,0 +1,98 @@
+package stan
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"github.com/cloudevents/sdk-go/pkg/binding"
+	"github.com/cloudevents/sdk-go/pkg/binding/format"
+	"github.com/cloudevents/sdk-go/pkg/binding/spec"
+	"github.com/nats-io/stan.go"
+)
+
+type Message struct {
+	*stan.Msg
+	ManualAcks bool
+}
+
+type MessageOption func(*Message) error
+
+func WithManualAcks() MessageOption {
+	return func(m *Message) error {
+		m.ManualAcks = true
+		return nil
+	}
+}
+
+// By the specification, all NATS message payloads must be in the JSON event format.
+// If this isn't the case then this function will return an error
+func NewMessage(msg *stan.Msg, opts ...MessageOption) (*Message, error) {
+	raw := make(map[string]json.RawMessage)
+
+	if err := json.Unmarshal(msg.Data, &raw); err != nil {
+		return nil, err
+	}
+
+	if _, err := spec.VS.FindVersion(versionExtractor(raw)); err != nil {
```

Comment:
Same here, any reason why you do that?

Reply:
For the reasons above, let's keep a single thread and resolve this comment